client.go

```diff
@@ -75,40 +75,38 @@ func (c *basicAuthClient) DoWithBody(req *http.Request, r io.ReadSeeker) (*http.
 	return c.client.DoWithBody(req, r)
 }
 
-// UbuntuSSOOAuthVisitWebPage returns a function that can be used with
-// httpbakey.Client.VisitWebPage to perform an OAuth login interaction.
-func UbuntuSSOOAuthVisitWebPage(client *http.Client, tok *usso.SSOData) func(u *url.URL) error {
-	return func(u *url.URL) error {
-		return ussoOAuthVisit(client, tok, u)
-	}
-}
-
-func ussoOAuthVisit(client *http.Client, tok *usso.SSOData, u *url.URL) error {
+// LoginMethods returns information about the available login methods
+// for the given URL, which is expected to be a URL as passed to
+// a VisitWebPage function during the macaroon bakery discharge process.
+func LoginMethods(client *http.Client, u *url.URL) (*params.LoginMethods, error) {
```

Comment:
Thanks for this, I like what you've done. I'm in two minds about this one, but maybe @rogpeppe will have an opinion. You could make the signature `LoginMethods(client ..., u ..., v interface{}) error` and unmarshal into v. that would have the advantage that applications could only look for interfaces they care about, and things like embedded juju need not pollute params.LoginMethods for IDPs that are rarely configured. Or you could just leave it.